File: BOJ/python2/10828.py
n = int(raw_input())
S = []
for _ in xrange(n):
    A = raw_input().split()
    if A[0] == 'push':
        S.append(int(A[1]))
    elif A[0] == 'pop':
        if not S: print -1
        else: print S.pop()
    elif A[0] == 'size':
        print len(S)
    elif A[0] == 'empty':
        print int(len(S) == 0)
    elif A[0] == 'top':
        if not S: print -1
        else: print S[-1]

File: baekjoon/10828.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    ios::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);
    int N;
    cin >> N;
    stack<int> st;
    while (N--){
        string s;
        cin >> s;
        if (s == "push"){
            int x;
            cin >> x;
            st.push(x);
        } else if (s == "top"){
            if (st.empty()) {
                cout << "-1\n";
            } else {
                cout << st.top() << '\n';
            }
        } else if (s == "size"){
            cout << st.size() << '\n';
        } else if (s == "pop"){
            if (st.empty()){
                cout << "-1\n";
            } else {
                cout << st.top() << '\n';
                st.pop();
            }
        } else if (s == "empty") {
            cout << (st.empty()) << '\n';
        } else assert(false);
    }
    return 0;
}